### PR TITLE
fix: validate api key

### PR DESCRIPTION
**Pull request by himself65**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama_index/embeddings/openai.py [hunk 1]
```diff
@@ -304,6 +304,14 @@ def __init__(
         self._aclient = None
         self._http_client = http_client
 
+    @validator("api_key", pre=True)
+    def _validate_api_key(cls, v: Optional[str]) -> Optional[str]:
```
Comment (hatianzhang) [inline llama_index/embeddings/openai.py:308]: is this function being used somewhere

Comment (himself65) [reply llama_index/embeddings/openai.py:311]: I think no

Comment (logan-markewich) [reply llama_index/embeddings/openai.py:311]: @hatianzhang the decorator hooks it into pydantic, it will be called automatically 👍🏻

**Review by logan-markewich — APPROVED** [review]

File: llama_index/embeddings/openai.py (hunk 1 above)
Comment (logan-markewich) [reply llama_index/embeddings/openai.py:311]: oh crap -- I should not have merged this 😓 This will break azure

**Pull request merged**